core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -483,6 +467,16 @@ public void start(Boolean isPrimary) throws IOException {
         }
       }
 
+      // Initialize the ufs manager from the mount table.
```

Comment:
Why is this block needed now, and not before?

Reply:
Before adding a mount point to the mount table, we first add it to the ufs manager and use the ufs manager to validate the mount attempt. This means we need to update the ufs manager before calling `applyAndJournal`, so we can't update the ufs manager as a part of `applyAndJournal`. We update it here instead. This is also where we would take care of things like setting all `IN_PROGRESS` persistence states to `NOT_PERSISTED`

Reply:
I think ideally the ufs manager wouldn't need to be manually kept in sync with the mount table. I'd prefer to leave that to a followup PR since it requires some refactoring